src/Http/Routing/test/UnitTests/Builder/DelegateEndpointRouteBuilderExtensionsTest.cs

```diff
@@ -176,6 +176,178 @@ public async Task MapGetWithoutRouteParameter_BuildsEndpointWithQuerySpecificBin
             Assert.Equal(41, httpContext.Items["input"]);
         }
 
+        [Fact]
+        public void MapGet_ThrowsWithImplicitFromBody()
+        {
+            var builder = new DefaultEndpointRouteBuilder(new ApplicationBuilder(new EmptyServiceProvider()));
+            var ex = Assert.Throws<InvalidOperationException>(() => builder.MapGet("/", (Todo todo) => { }));
+            Assert.Contains("Body was inferred but the method does not allow inferred body parameters.", ex.Message);
+            Assert.Contains("Did you mean to register the \"Body (Inferred)\" parameter(s) as a Service or apply the [FromService] or [FromBody] attribute?", ex.Message);
+        }
+
+        [Fact]
+        public void MapDelete_ThrowsWithImplicitFromBody()
+        {
+            var builder = new DefaultEndpointRouteBuilder(new ApplicationBuilder(new EmptyServiceProvider()));
+            var ex = Assert.Throws<InvalidOperationException>(() => builder.MapDelete("/", (Todo todo) => { }));
+            Assert.Contains("Body was inferred but the method does not allow inferred body parameters.", ex.Message);
+            Assert.Contains("Did you mean to register the \"Body (Inferred)\" parameter(s) as a Service or apply the [FromService] or [FromBody] attribute?", ex.Message);
+        }
+
+        public static object[][] NonImplicitFromBodyMethods
+        {
+            get
+            {
+                return new[]
+                {
+                    new[] { HttpMethods.Delete },
+                    new[] { HttpMethods.Connect },
+                    new[] { HttpMethods.Trace },
+                    new[] { HttpMethods.Get },
+                    new[] { HttpMethods.Head },
+                    new[] { HttpMethods.Options },
+                };
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(NonImplicitFromBodyMethods))]
+        public void MapVerb_ThrowsWithImplicitFromBody(string method)
+        {
+            var builder = new DefaultEndpointRouteBuilder(new ApplicationBuilder(new EmptyServiceProvider()));
+            var ex = Assert.Throws<InvalidOperationException>(() => builder.MapMethods("/", new[] { method }, (Todo todo) => { }));
+            Assert.Contains("Body was inferred but the method does not allow inferred body parameters.", ex.Message);
+            Assert.Contains("Did you mean to register the \"Body (Inferred)\" parameter(s) as a Service or apply the [FromService] or [FromBody] attribute?", ex.Message);
+        }
+
+        [Fact]
+        public void MapGet_ImplicitFromService()
+        {
+            var builder = new DefaultEndpointRouteBuilder(new ApplicationBuilder(new ServiceCollection().AddSingleton<Todo>().BuildServiceProvider()));
+            _ = builder.MapGet("/", (Todo todo) => { });
```

Comment:
I just realized that Todo was a service. I would add a second TodoService type.

Reply:
Done.